sql/npc_list.sql

```diff
@@ -26,7 +26,7 @@ CREATE TABLE `npc_list` (
   `pos_z` float(7,3) NOT NULL DEFAULT 0.000,
   `flag` int(10) unsigned NOT NULL DEFAULT 0,
   `speed` tinyint(3) unsigned NOT NULL DEFAULT 40,
-  `speedsub` tinyint(3) unsigned NOT NULL DEFAULT 40,
+  `speedbase` tinyint(3) unsigned NOT NULL DEFAULT 40,
```

Comment:
Capture tools use speedsub. I would keep the original name unless the change is necesary.

Reply:
Yeah not necessary, just better aligns with what it actually is and XIPackets repo documentation.

Something else I noticed somewhat related to this is that in incoming packet 0x00E at position 0x25 the server sends the entity model size, which I think is used for calculating melee range. Right now I don't think any of the capture tools parse this specifically and I think most model sizes we are using are mostly just made up? (not sure on that) So I would like to see that field added to whatever capture tool is capturing this stuff. I assume the same tool is what's using speedsub, so idk if this would be a good time to change it, or keep speedsub and add model size, or whatever you guys wanna do (not sure who maintains that stuff).

I do have a half finished packet parser that can go back over old caps and can pull that model info out as well, but would be nice to have it parsed out along with the other stuff. It's no issue removing that commit here though if that's what you wanna do. Let me know, I'll probably keep this in draft for a few days until I can get on retail anyway.

Reply:
What kind of weapon is that? If we disregard the player model, which I think is done in some or all cases, then 8.3 - 2.8 = 5.5, which looks like the sweet spot for guns. I'm curious what your max melee is too.

Reply:
Oh, that's an old pic when I was testing ranged distance. I think it was crossbow?

Didn't check melee distance at that time

Reply:
What plugin is that and where can I get it? After some preliminary testing I'm seeing some very interesting results.

```
Model size: 0.3
Max Melee (Taru): 2.6
Max Melee (Hume): 2.7

Model size: 1.1
Max Melee (Taru): 3.4
Max Melee (Hume): 3.5
Max Melee (Galka): 3.9

Model size: 2.7
Max Melee (Taru): 5.0
Max Melee (Hume): 5.1
Max Ranged (Both): 25.0

Model size: 3.4
Max Melee (Galka): 6.2
```

This is on flat ground. Being on raised ground seems to increase this, so vertical distance doesn't seem to be a factor (probably some vertical limit instead of being factored into distance). From your screenshot it looks like Hume is 0.4, I'm guessing Taru is 0.3, which would make melee distance something like `enemy model size + player model size + 2.0`. Ranged seems to be point to point 25.0 with no model size adjustments. Model size also doesn't seem to be sent in every packet.

Reply:
I just hacked up ashita's distance

```
    local player = GetPlayerEntity()
    -- Ensure there is a target..
    local target = GetEntity(AshitaCore:GetMemoryManager():GetTarget():GetTargetIndex(0));
    if (target ~= nil) then
        distance.font.text = ('%.1f:%.1f; TargetSize: %.1f, YourSize: %.1f, TargetModelSize:%f'):fmt(math.sqrt(target.Distance), math.sqrt(target.Distance)-target.ModelHitboxSize-player.ModelHitboxSize, target.ModelHitboxSize, player.ModelHitboxSize, target.ModelSize);
    else
        distance.font.text = '';
    end
```
note that this is junk and shouldn't be used
`math.sqrt(target.Distance)-target.ModelHitboxSize-player.ModelHitboxSize`

I think windower has an equivalent for the model hitbox size but not 100% sure. Note that model size is independent of the hitbox size, theoretically SE could send any arbitrary number for either

Reply:
I'm not sure if Windower tracks this, but it makes sense that Ashita does. [It's not in their fields file](https://github.com/Windower/Lua/blob/45db20d088daf72d97e358c3f4b79861d9e3d92c/addons/libs/packets/fields.lua#L1472), which is what packetviewer uses, but they do have some model stuff in their wiki. I'll have to look later. I was just using packetviewer to look at the update packet.

> Note that model size is independent of the hitbox size, theoretically SE could send any arbitrary number for either

I think this is the hitbox size. [This is what I'm looking at](https://github.com/atom0s/XiPackets/tree/main/world/server/0x000E#g), which would be ModelHitboxSize assuming Ashita would use the same wording. Considering the model sizes + 2 melee distance holds across multiple different mobs showing different model sizes I think I'm on the right track here. I know there's some other model size related things too, like S/M/L players and a bit that makes mobs bigger. I wonder if any of that is factored in as well. I'm honestly surprised player race makes a difference in melee range.

Reply:
To the original topic, I reverted the speedsub -> speedbase change. This model size stuff will need additional captures. NPCLogger saves the raw packet so can probably reparse a lot of that stuff too.

Reply:
> I think this is the hitbox size. [This is what I'm looking at](https://github.com/atom0s/XiPackets/tree/main/world/server/0x000E#g), which would be ModelHitboxSize assuming Ashita would use the same wording.
Yup, that's the one.